### ISSUE TITLE
Test failure: MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/Templating#748]
This test [fails](http://aspnetci/viewLog.html?buildId=557861&buildTypeId=Lite_UniverseTest) occasionally with the following error:
```
System.AggregateException : One or more errors occurred. (An error occurred while sending the request.)
---- System.Net.Http.HttpRequestException : An error occurred while sending the request.
-------- System.IO.IOException : Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host..
------------ System.Net.Sockets.SocketException : An existing connection was forcibly closed by the remote host.
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Templates.Test.Helpers.AspNetProcess.AssertStatusCode(String requestUrl, HttpStatusCode statusCode, String acceptContentType) in /_/test/Templates.Test/Helpers/AspNetProcess.cs:line 168
   at Templates.Test.Helpers.AspNetProcess.AssertOk(String requestUrl) in /_/test/Templates.Test/Helpers/AspNetProcess.cs:line 152
   at Templates.Test.MvcTemplateTest.MvcTemplate_NoAuthImpl(String targetFrameworkOverride, String languageOverride, Boolean noHttps) in /_/test/Templates.Test/MvcTemplateTest.cs:line 78
   at Templates.Test.MvcTemplateTest.MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate() in /_/test/Templates.Test/MvcTemplateTest.cs:line 24
----- Inner Stack Trace -----
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithNtConnectionAuthAsync(HttpConnection connection, HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
----- Inner Stack Trace -----
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.FillAsync()
   at System.Net.Http.HttpConnection.ReadNextResponseHeaderLineAsync(Boolean foldedHeadersAllowed)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
----- Inner Stack Trace -----
------- Stdout: -------
==> C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe new mvc --debug:custom-hive "C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\.templateengine" --target-framework-override net461 -lang F# [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22]
The template "ASP.NET Core Web App (Model-View-Controller)" was created successfully.
This template contains technologies from parties other than Microsoft, see https://aka.ms/aspnetcore-template-3pn-210 for details.
Processing post-creation actions...
Running 'dotnet restore' on C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22\AspNet.Template.468553c732514d0886274b3e73542f22.fsproj...
  Restoring packages for C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22\AspNet.Template.468553c732514d0886274b3e73542f22.fsproj...
  Generating MSBuild file C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22\obj\AspNet.Template.468553c732514d0886274b3e73542f22.fsproj.nuget.g.props.
  Generating MSBuild file C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22\obj\AspNet.Template.468553c732514d0886274b3e73542f22.fsproj.nuget.g.targets.
  Restore completed in 857.92 ms for C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22\AspNet.Template.468553c732514d0886274b3e73542f22.fsproj.
Restore succeeded.
Building ASP.NET application...
==> C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe build --no-restore -c Debug [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22]
Microsoft (R) Build Engine version 15.8.166+gd4e8d81a88 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.
  AspNet.Template.468553c732514d0886274b3e73542f22 -> C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22\bin\Debug\net461\AspNet.Template.468553c732514d0886274b3e73542f22.exe
Build succeeded.
    0 Warning(s)
    0 Error(s)
Time Elapsed 00:00:07.64
Running ASP.NET application...
==> C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22\bin\Debug\net461\AspNet.Template.468553c732514d0886274b3e73542f22.exe  [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp3.0\TestTemplates\AspNet.Template.468553c732514d0886274b3e73542f22]
Waiting until ASP.NET application is accepting connections...
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using 'C:\Users\asplab\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
Hosting environment: Development
Content root path: C
...
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on master.

CC @mkArtakMSFT

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
MvcTemplate_WithIndividualAuth_NoHttpsSetToTrue_UsesHttps_NetFramework [failed with about the same error](http://aspnetci/viewLog.html?buildId=557861&buildTypeId=Lite_UniverseTest) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
MvcTemplate_NoAuth_NoHttps_Works_NetFramework_ForFSharpTemplate [failed with about the same error](http://aspnetci/viewLog.html?buildId=558150&buildTypeId=Lite_UniverseTest) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by mkArtakMSFT** [comment]
@ryanbrandenburg, can you please look into this? Thanks!

**Comment by ryanbrandenburg** [comment]
This is the same thing as aspnet/Home#3544.

**Comment by aspnet-hello** [comment]
There were 14 failures [with about the same error](http://aspnetci/viewLog.html?buildId=558745&buildTypeId=Lite_UniverseTest) on master:
- MvcTemplate_NoAuth_Works_NetFramework_ForDefaultTemplate
- RazorPagesTemplate_NoAuth_Works_NetFramework
- MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate
- MvcTemplate_WithIndividualAuth_NoHttpsSetToTrue_UsesHttps_NetFramework
- RazorPagesTemplate_NoAuth_NoHttps_Works_NetFramework
- AngularTemplate_NoHttps_Works_NetFramework
- MvcTemplate_IndividualAuth_Works_NetFramework
- EmptyWebTemplate_Works_NetFramework
- WebApiTemplate_Works_NetFramework
- RazorPagesTemplate_IndividualAuth_Works_NetFramework
- (And 4 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 14 failures [with about the same error](http://aspnetci/viewLog.html?buildId=559515&buildTypeId=Lite_UniverseTest) on master:
- MvcTemplate_NoAuth_NoHttps_Works_NetFramework_ForDefaultTemplate
- AngularTemplate_NoHttps_Works_NetFramework
- EmptyWebTemplate_Works_NetFramework
- RazorPagesTemplate_WithIndividualAuth_NoHttpsSetToTrue_UsesHttps_NetFramework
- WebApiTemplate_Works_NetFramework
- MvcTemplate_IndividualAuth_Works_NetFramework
- RazorPagesTemplate_IndividualAuth_Works_NetFramework
- RazorPagesTemplate_NoAuth_Works_NetFramework
- MvcTemplate_NoAuth_Works_NetFramework_ForDefaultTemplate
- AngularTemplate_Works_NetFramework
- (And 4 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 14 failures [with about the same error](http://aspnetci/viewLog.html?buildId=560636&buildTypeId=Lite_UniverseTest) on master:
- EmptyWebTemplate_Works_NetFramework
- RazorPagesTemplate_NoAuth_NoHttps_Works_NetFramework
- MvcTemplate_WithIndividualAuth_NoHttpsSetToTrue_UsesHttps_NetFramework
- RazorPagesTemplate_WithIndividualAuth_NoHttpsSetToTrue_UsesHttps_NetFramework
- RazorPagesTemplate_IndividualAuth_Works_NetFramework
- MvcTemplate_NoAuth_Works_NetFramework_ForDefaultTemplate
- MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate
- AngularTemplate_Works_NetFramework
- MvcTemplate_IndividualAuth_Works_NetFramework
- MvcTemplate_NoAuth_NoHttps_Works_NetFramework_ForDefaultTemplate
- (And 4 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 14 failures [with about the same error](http://aspnetci/viewLog.html?buildId=561382&buildTypeId=Lite_UniverseTest) on master:
- MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate
- AngularTemplate_NoHttps_Works_NetFramework
- WebApiTemplate_Works_NetFramework
- RazorPagesTemplate_NoAuth_NoHttps_Works_NetFramework
- RazorPagesTemplate_WithIndividualAuth_NoHttpsSetToTrue_UsesHttps_NetFramework
- MvcTemplate_IndividualAuth_Works_NetFramework
- MvcTemplate_NoAuth_NoHttps_Works_NetFramework_ForFSharpTemplate
- MvcTemplate_WithIndividualAuth_NoHttpsSetToTrue_UsesHttps_NetFramework
- AngularTemplate_Works_NetFramework
- EmptyWebTemplate_Works_NetFramework
- (And 4 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
This was resolved by the work done in aspnet/Home#3544.
